controllers/confidential_handler.go

```diff
@@ -18,61 +38,136 @@ import (
-				imageConfigMapData := map[string]string{"CONFIDENTIAL_COMPUTE_ENABLED": "yes"}
-				if err := updateConfigMap(r.Client, r.Log, ig.getImageConfigMapName(), OperatorNamespace, imageConfigMapData); err != nil {
-					return err
-				}
+			if ig.isImageIDSet() {
+				r.Log.Info("Image ID is already set, skipping confidential image configuration")
 			} else {
-				// Patch ImageConfigMap.
-				imageConfigMapData := map[string]string{"CONFIDENTIAL_COMPUTE_ENABLED": "no"}
-				if err := updateConfigMap(r.Client, r.Log, ig.getImageConfigMapName(), OperatorNamespace, imageConfigMapData); err != nil {
-					if k8serrors.IsNotFound(err) {
-						// Nothing to do, feature is disabled and configMap doesn't exist.
-					} else {
+				if state == Enabled {
+					// Create ImageConfigMap, if it doesn't exist already.
+					if err := ig.createImageConfigMapFromFile(); err != nil {
+						return err
+					}
+
+					// Patch ImageConfigMap.
+					imageConfigMapData := map[string]string{"CONFIDENTIAL_COMPUTE_ENABLED": "yes"}
+					if err := updateConfigMap(r.Client, r.Log, ig.getImageConfigMapName(), OperatorNamespace, imageConfigMapData); err != nil {
 						return err
 					}
+				} else {
+					// Patch ImageConfigMap.
+					imageConfigMapData := map[string]string{"CONFIDENTIAL_COMPUTE_ENABLED": "no"}
+					if err := updateConfigMap(r.Client, r.Log, ig.getImageConfigMapName(), OperatorNamespace, imageConfigMapData); err != nil {
+						if k8serrors.IsNotFound(err) {
+							// Nothing to do, feature is disabled and configMap doesn't exist.
+						} else {
+							return err
+						}
+					}
 				}
 			}
 		}
-	}
 
-	// peer pods config
+		// Patch peer pods configMap, if it exists.
+		var peerpodsCMData map[string]string
+		if state == Enabled {
+			peerpodsCMData = map[string]string{"DISABLECVM": "false"}
+		} else {
+			peerpodsCMData = map[string]string{"DISABLECVM": "true"}
+		}
+		if err := updateConfigMap(r.Client, r.Log, peerpodsCMName, OperatorNamespace, peerpodsCMData); err != nil {
+			if k8serrors.IsNotFound(err) {
+				// When feature is Enabled: ConfigMap doesn't exist yet, will try again at the next reconcile run.
+				// Else: Nothing to do, feature is disabled and configMap doesn't exist.
+			} else {
+				return err
+			}
+		}
+	}
 
-	// Patch peer pods configMap, if it exists.
-	var peerpodsCMData map[string]string
+	// Handle kata-cc runtime classes for baremetal environments
 	if state == Enabled {
-		peerpodsCMData = map[string]string{"DISABLECVM": "false"}
+		r.Log.Info("Creating kata-cc runtime classes for confidential containers")
+
+		handler, nodeLabel, err := r.computeTEEHandlerAndLabel()
+		if err != nil {
+			r.Log.Info("TEE detection failed", "err", err)
+			return fmt.Errorf("TEE detection failed: %w", err)
+		}
+
+		// Create kata-cc runtime class restricted to the detected TEE subset
+		err = r.createRuntimeClass(kataCCRuntimeClassName, kataCCRuntimeClassCpuOverhead, kataCCRuntimeClassMemOverhead, handler, nodeLabel)
```

Comment:
in addition to CPU and Mem overhead, we consume "key ID" overhead resources for each CVM.

Reply:
I'm not sure if we should handle this inside the RuntimeClass or would it need to be a different mechanism. IIUC, it seems like a different type of resource.

Since this isn't handled today, I think it should be addressed in a separate PR.
Could you expand a bit on this key ID overhead?